app/routines/update_user_contact_info.rb

```diff
@@ -101,6 +101,13 @@ def call
                            :other_school_type
                          end
 
+      if school != user.school.salesforce_id
+        user.school = School.find_by(salesforce_id: user.school.salesforce_id)
```

Comment:
This is a no-op but it hits the database, which is not great.
`user.school` should already return the thing you are setting it to.
I'm thinking what was here before (the `else` part of the if statement) would already fix mismatched schools. You could put a check inside the `else` to log if a mismatch happened (like actually inside the `else`, not `elsif`).

It won't fix the mismatch if `school.nil?` which will get logged by the other thing.
This case can maybe be greatly reduced by ensuring this routine always runs right after the routine that syncs Schools from Salesforce.

Reply:
I see what you are saying now - it took a little walkthrough of the code. I'm going to take this out - it sounds like I need to rethink the problem we saw (which might have been from a legacy process).
I'll investigate more.